plugins/mac-capture/mac-screen-capture.m

```diff
@@ -835,24 +835,52 @@ static bool build_window_list(struct screen_capture *sc,
 	obs_property_t *window_list = obs_properties_get(props, "window");
 	obs_property_list_clear(window_list);
 
-	[sc->shareable_content.windows enumerateObjectsUsingBlock:^(
-					       SCWindow *_Nonnull window,
-					       NSUInteger idx
-					       __attribute__((unused)),
-					       BOOL *_Nonnull stop
-					       __attribute__((unused))) {
+	NSArray<SCWindow *> *filteredWindows;
+	filteredWindows = [sc->shareable_content.windows
+		filteredArrayUsingPredicate:
+			[NSPredicate predicateWithBlock:^BOOL(
+					     SCWindow *window,
+					     NSDictionary *bindings
+					     __attribute__((unused))) {
+				NSString *app_name =
+					window.owningApplication.applicationName;
+				NSString *title = window.title;
+				if (!sc->show_empty_names) {
+					if (app_name == NULL || title == NULL) {
+						return false;
+					} else if ([app_name
+							   isEqualToString:@""] ||
+						   [title isEqualToString:@""]) {
+						return false;
+					}
+				}
+				return true;
+			}]];
+
+	NSArray<SCWindow *> *sortedWindows;
+	sortedWindows =
+		[filteredWindows sortedArrayUsingComparator:^NSComparisonResult(
+					 SCWindow *window, SCWindow *other) {
+			NSComparisonResult appNameCmp =
+				[window.owningApplication.applicationName
```

Comment:
Per Jim's comment, please add `options: NSCaseInsensitiveSearch` to the `compare` message.